### ISSUE TITLE
[a11y] Banners have some live region conflicts

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#790]
## Issue summary

Some [Banner](https://polaris.shopify.com/components/feedback-indicators/banner#navigation) examples show using live regions but demonstrate conflicting parings between `role` values and `aria-live` politeness settings, or aren't read due to other quirks.

- The paired attributes cause double reading in VoiceOver on iOS, which is made more verbose by the use of `aria-labelledby` and `aria-describedby`.
- NVDA doesn't read `role="..."` content if the container with `role="..."` doesn't already exist in the DOM when the content loads. Since the container and the content are appended to the DOM at the same time, the content isn't read. [This appears to be a won't fix in NVDA, for reasons.](https://github.com/nvaccess/nvda/issues/5803#issuecomment-194604503)
- The yellow warning banners and red critical banners use `role="alert"` and `aria-live="polite"`. By default, [`role="alert"`](https://www.w3.org/TR/wai-aria-1.1/#alert) implicitly has `aria-live="assertive"` since it is meant for situations where the user should quickly act to address an error or warning. [`role="status"`](https://www.w3.org/TR/wai-aria-1.1/#status), on the other hand, implicitly has `aria-live="polite"` since it is meant for announcements that are nice to know but not critical to using the page or app right now.

#### Status (green or gray) banner example (HTML)

```
<div class="Polaris-Banner Polaris-Banner--withinPage" tabindex="0" role="status" aria-live="polite" aria-labelledby="Banner16Heading" aria-describedby="Banner16Content">
<!-- ... -->
</div>
```

#### Warning (yellow) banner example (HTML)

```
<div class="Polaris-Banner Polaris-Banner--statusWarning Polaris-Banner--withinPage" tabindex="0" role="alert" aria-live="polite" aria-labelledby="Banner7Heading" aria-describedby="Banner7Content">
<!-- ... -->
</div>
```

#### Critical (red) banner example (HTML)

```
<div class="Polaris-Banner Polaris-Banner--statusCritical Polaris-Banner--withinPage" tabindex="0" role="alert" aria-live="polite" aria-labelledby="Banner19Heading" aria-describedby="Banner19Content">
<!-- ... -->
</div>
```

## Steps to reproduce

1. Navigate to [Banner](https://polaris.shopify.com/components/feedback-indicators/banner#navigation) using any supported browser/screen reader combination.
1. Swap out the different examples to hear how they are read when the banners appear.
1. Examine the HTML output of the`Banner with footer call to action`, `Warning banners`, `Critical banners`, and `Banner in a modal` examples in the web inspector.

### Expected

- Reading will be clear and consistent.
- ARIA roles and properties will be used as specified to provide support as consistently as possible.

### Actual

- VoiceOver on iOS repeats `aria-live` messages, and NVDA does not read the `role="status"` examples when they are added to the page.
- `role="alert"` is used with `aria-live="polite".

## Recommendations

- When `role="status"` or `role="alert"` is used, don't add an additional  `aria-live` attribute, since those roles implicitly include one, and doubling up can cause double reading or politeness conflicts. 
- To support NVDA, make sure it's possible to dynamically add the banner content after the banner container with `role="..."` is loaded. This might be a case of documentation instead of implementation, however (see also #1732).

#### Status banner example (HTML)

```
<div class="Polaris-Banner Polaris-Banner--withinPage" tabindex="0" role="status" aria-labelledby="Banner16Heading" aria-describedby="Banner16Content">
<!-- dynamically append stuff here after the role="status" container is loaded in the DOM for full NVDA support -->
</div>
```

#### Warning banner example (HTML)

```
<div class="Polaris-Banner Polaris-Banner--statusWarning Polaris-Banner--withinPage" tabindex="0" role="alert" aria-labelledby="Banner7Heading" aria-describedby="Banner7Content">
<!-- ... -->
</div>
```

## Specifications

- **Component:** Banner
- **WCAG SCs:** [4.1.3 Status Messages (Level AA)](https://www.w3.org/TR/WCAG21/#status-messages)
- **Testing done with:**
  - VoiceOver/Safari/macOS
  - VoiceOver/Safari/iOS
  - NVDA/Firefox/Windows 10

**Comment by dpersing** [comment]
This issue suffers from the problem where the live region is being applied to the page along with the content at the same time, so it is not always conveyed clearly, so it would benefit from some discussion.

**Comment by dpersing** [comment]
Closing this issue for a couple of reasons:

- With retesting (iOS, Android, macOS, and Windows), the duplicate reading is really only a problem for VO, which duplicates the text because of `aria-labelledby`.
- We recently made changes to leverage the `aria-live` attribute when content _shouldn't_ be read out to users (see https://github.com/Shopify/polaris-react/pull/1719)
